### PR TITLE
ref(project): Skip serializing default fields

### PR DESCRIPTION
**Pull request by jjbayer**
Some fields that are not being sent by Sentry when they are empty are serialized by Relay. Align with Sentry and do not serialize these fields if they have default values.

**Why though?**

* The main reason is that https://github.com/getsentry/relay/pull/1860 introduced a validation function that can be used Sentry-side to check if Sentry accidentally produces any fields that Relay cannot use. This has already surfaced some inconsistencies (see https://github.com/getsentry/sentry/pull/45124), but it requires that Sentry and Relay produce the same fields.
* Reduced payload size when sending configs to downstream Relays.

### CODE REVIEW

File: relay-general/src/pii/config.rs [hunk 1]
```diff
@@ -121,7 +127,7 @@ pub struct MultipleRule {
     /// A reference to other rules to apply
     pub rules: Vec<String>,
     /// When set to true, the outer rule is reported.
-    #[serde(default)]
+    #[serde(default, skip_serializing_if = "is_flag_default")]
```
Comment (jjbayer) [inline relay-general/src/pii/config.rs:130]: `#[serde(default)]` has been here for 4 years, so every Relay should be able to handle the field's absence. I checked the age of the `default` functionality for the other fields as well.

**Review by olksdr — APPROVED** [review]
lgtm

File: relay-general/src/pii/config.rs [hunk 2]
```diff
@@ -199,23 +205,29 @@ pub struct RuleSpec {
...
 pub struct Vars {
     /// The default secret key for hashing operations.
-    #[serde(default)]
+    #[serde(default, skip_serializing_if = "Option::is_none")]
     pub hash_key: Option<String>,
 }
 
+impl Vars {
+    fn is_empty(&self) -> bool {
+        self.hash_key.is_none()
```
Comment (iker-barriocanal) [inline relay-general/src/pii/config.rs:214]: Should we also consider post-trimmed empty strings as empty vars? In the tests below (from the diff), I only see `hash_key` being `null`.

**Review by iker-barriocanal — APPROVED** [review]

File: relay-general/src/pii/config.rs (hunk 2 above)
Comment (jjbayer) [reply relay-general/src/pii/config.rs:214]: You might have more context than me here, when would `hash_key` be an empty string?
